### PR TITLE
Add documentation for threads & buttons

### PR DESCRIPTION
**Pull request by jgayfer**
With the upcoming changes to Apollo around threads and buttons, our documentation will need to be updated!

### Todo

- [x] purge on delete
- [x] deleting events
- [x] editing events
- [x] event reminders
- [x] event threads command
- [x] thread reminders command

### CODE REVIEW

**Comment by jgayfer** [comment]
@reiichi001 Ready for a review! Though let's hold off on merging until the update is live.

File: docs/event_channels.md [hunk 1]
```diff
@@ -58,6 +58,7 @@ channel. This limit is increased to **25** if there is an active Premium members
 ## Purge non event messages
 
 By default non event messages (that aren't pinned) are purged from an event
-channel when a new event is posted in the channel.
+channel when a new event is posted or deleted. The goal of this feature is to
+maintain an event channel as a place for events only, free of distraction.
```
Comment (reiichi001) [inline docs/event_channels.md:62]: One thing to consider is "how will this affect threads?" 

Would purge also archive/expire any old event threads? Would we just be letting those expire as per the user's server's rules?

File: docs/reminders.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Reminders
```
Comment (reiichi001) [inline docs/reminders.md:1]: We will want to make sure that Apollo checks that it has the right thread permissions.

We should also probably run a check that users accessing the channel have permission to see/send messages in threads.

File: docs/event_channels.md (hunk 1 above)
Comment (jgayfer) [reply docs/event_channels.md:62]: Purge has no effect on threads directly. If we delete a message that's threaded, the thread will live on behind the scenes, we just won't see a parent message any longer.

I'm not sure if this is worth documenting. Thoughts?

File: docs/reminders.md (hunk 2 above)
Comment (jgayfer) [reply docs/reminders.md:1]: We do check thread permissions! If Apollo is not able to create or send a message in a thread, the event organizer is notified via DM of the missing permissions to ensure a reminder is sent in the future.

I don't think there's a permission that regular attendees need (or that we should check). As long as they can view the channel, they should be good. Even if they don't have permission to create threads or send messages in them, they should still be able to view the public thread, which is all that really matters for reminders.

File: docs/event_channels.md (hunk 1 above)
Comment (jgayfer) [reply docs/event_channels.md:62]: Though that reminds me if an _event_ specifically is deleted, and the thread was created by Apollo, we do manually archive the thread right away to clean up.

That might be worth documenting under event deletion.

Comment (reiichi001) [reply docs/event_channels.md:62]: To cover our bases, I think we'd benefit by telling the user what will or won't happen to threads, even if it's the default Discord behavior. Threads are still new and changing as evidenced by Discord's upcoming permission changes for Threads. I think we can expect they're probably going to make bot makers miserable by revising Threads again once they've collected some more user feedback.

File: docs/reminders.md (hunk 2 above)
Comment (reiichi001) [reply docs/reminders.md:1]: Right now, it looks like Discord doesn't have a "view threads" permission, which I have to imagine is going to be part of "see channels / read messages" right now, but it's definitely something we should keep an eye out for. Even if that's just a comment in the codebase saying "if discord ever decides to lock thread viewer permissions, this is where we should check that a user can view threads." 

If we're not going to enforce that users will/should be able to send messages to a thread that they're effectively forced-added to, we should note that in the documentation. Something small like "be sure to check your server's permissions if you'd like your members to be able to send messages in threads that Apollo creates! Or not!"

**Review by reiichi001 — APPROVED** [review]
Added additional comments. Overall, the documentation looks good to me. I will always default to providing too much information to a user, especially on default expectations.

**Pull request merged**